### ISSUE TITLE
@Prompts Keyword for Structured and Reusable Prompt Templates

### ISSUE THREAD

**Issue by stevennevins** [getcursor/cursor#1432]
**Is your feature request related to a problem? Please describe.**

I often perform repetitive tasks, and to get the best results, I have to craft a descriptive prompt for the first generation. For subsequent tasks, I have to copy and edit a portion of the context. Often, I only edit a single line in the context of the subsequent tasks.

**Describe the solution you'd like**
Custom prompt templates accessible via the @Prompts keyword/command. @Prompts can be created within the Cursor editor and provide a structured prompt for commonly performed tasks.  

```
You are an AI assistant designed to perform a task.

Additional high-level context is helpful for the AI to perform the tasks to your specifications.

{{TASK_DESCRIPTION}}

Reiterate some context
```

I want to be able to create templates accessible with @Prompts and then provide context by named values to be inserted into these templates.  So, for example, if this template were named `Template`, then I would want to be able to do:

`@Prompts:@Template:@Task_Description:{{Task Description Inserted into Template}} `

**Comment by ypref** [comment]
+1

**Comment by dgokcin** [comment]
+1

**Comment by Jiayou-Chao** [comment]
You can use `/` to call prompt `edit`, it's just that you can't add your customized prompts.

**Comment by dgokcin** [comment]
@Jiayou-Chao  Yes, continue.dev has implemented this feature quite well. For reference, with the hope of implementing this in Cursor, I'm attaching the relevant documentation link. They are utilizing a JSON file located at ~/.continue/config.json for further configuration of their tool.

[Custom Commands](https://docs.continue.dev/customization/slash-commands#custom-commands-use-natural-language)


<img width="547" alt="image" src="https://github.com/getcursor/cursor/assets/33603535/0f9f6559-9c92-4e24-b5eb-e927eada6e29">

**Comment by 3choff** [comment]
+1

**Comment by iraklisg** [comment]
I do agree that Continue Dev has well implemented this feature. Related issue https://github.com/getcursor/cursor/issues/1019

**Comment by mduccc** [comment]
+1
